iroh-net/src/relay/http/client.rs

```diff
@@ -580,6 +588,53 @@ impl Actor {
     }
 
     async fn connect_0(&self) -> Result<(RelayClient, RelayClientReceiver), ClientError> {
+        // We determine which protocol to use for relays via the URL scheme: ws(s) vs. http(s)
+        let protocol = Protocol::from_url_scheme(&self.url);
+
+        let (reader, writer, local_addr) = match &protocol {
+            Protocol::Websocket => {
+                let (reader, writer) = self.connect_ws().await?;
+                let local_addr = None;
+                (reader, writer, local_addr)
+            }
+            Protocol::Relay => {
+                let (reader, writer, local_addr) = self.connect_derp().await?;
+                (reader, writer, Some(local_addr))
+            }
+        };
+
+        let (relay_client, receiver) =
+            RelayClientBuilder::new(self.secret_key.clone(), local_addr, reader, writer)
+                .build()
+                .await
+                .map_err(|e| ClientError::Build(e.to_string()))?;
+
+        if self.is_preferred && relay_client.note_preferred(true).await.is_err() {
+            relay_client.close().await;
+            return Err(ClientError::Send);
+        }
+
+        trace!("connect_0 done");
+        Ok((relay_client, receiver))
+    }
+
+    async fn connect_ws(&self) -> Result<(RelayConnReader, RelayConnWriter), ClientError> {
+        let mut dial_url = (*self.url).clone();
+        dial_url.set_path("/derp");
```

Comment:
should we use a different url here?

Reply:
I think we should look at relay paths again, and perhaps solve #2378 at the same time.
At the moment the setup is weird:
- There's a config option for `iroh-relay` to use a different path than `/derp`
- There's no way to configure the client to connect to somewhere else than `/derp`...

So at the moment the `iroh-relay` config option is available but de-facto must be set to the default value.
I think we should either make it hard-coded in the relay, or perhaps make the client configurable, or even make `RelayUrl` the source of truth for the path.

---

That all doesn't answer the Q :upside_down_face: 

I think, eventually we should just switch to `/relay` and run both the derp & ws protocols over that.
I don't see the advantage to using separate paths for them. Perhaps it makes blocking one protocol type easier or something if people have a reverse proxy configured? Is that worth it?

Reply:
> Today the actual route used in iroh-relay is governed by a config variable. 

We should probably drop this..no client works with this anyway..

Reply:
Yes, agreed, that was one of the options I was trying to convey with my first reply. I failed at being concise or focused though :sweat_smile: 

I'll create another PR on top of this that does exactly that!